kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CompletableFutures.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+public class CompletableFutures {
+
+    /**
+     * Sometimes we want to control which Executor a Future completes on, for example to control which
+     * thread we update mutable state from.
+     * @return the input future if it is already complete, else a CompletableFuture completed with the result of input future on Executor
+     */
+    public static <T> CompletableFuture<T> switchExecutorIfDeferred(CompletableFuture<T> future, Executor executor) {
```

Comment:
1. This could be implemented using `CompletionStage` rather than needing a `CompletableFuture` parameter.
2. Given the way you're using this, I wonder if this should be a non-static method, and have an instance field for `Executor`. That way you could instantiate a `CompletableFutures` once in the filter and pass it around instead of the executor. You'd probably want to rename both this class and method. This results in an abstraction which we may in the future want to consider elevating to the filter context itself.

Reply:
done